File: C++/Array/Reverse.cpp
#include <iostream>
#include <vector>
using namespace std;

void reverse(vector<int> &arr){
	int n = arr.size();
	vector<int> temp(n);
	//append reverse into temp array
	for (int i = 0; i < n; i++) {
		temp[i] = arr[n-1-i];
	}
	//make temp arr into real array
	for (int i = 0; i < n; i++) {
		arr[i] = temp[i];
	}

}

int main() {
	vector<int> arr = {1, 4, 5, 7, 8, 13};
	reverse(arr);

	for (int num:arr){
		cout << num << " ";
	}
	cout << "\n";
	return 0;
}	

File: python/math/Reverse.py
class Reverse:
    def __init__(self, arr):
        self.arr = arr
    
    def reverse(self, arr):
        n = len(self.arr)
        temp = []
        for i in range(n):
            temp.append(self.arr[n-i-1])
            
        for i in range(n):
            self.arr[i] = temp[i]

if __name__ == "__main__":
    arr = [1, 32, 24, 56, 78, 45]
    rv = Reverse(arr)
    rv.reverse(arr)
    for i in arr:
        print(i, end = " ")
    print("")
